src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -40,34 +41,42 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             }
 
             var cancellationToken = context.CancellationToken;
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            var identifier = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start,
-                token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
-            if (identifier == default)
+            var localFunction = await TryGetSelectedLocalFunctionAsync(document, context.Span, cancellationToken).ConfigureAwait(false);
+            if (localFunction == default)
             {
                 return;
             }
 
-            if (context.Span.Length > 0 &&
-                context.Span != identifier.Span)
+            if (!localFunction.Parent.IsKind(SyntaxKind.Block, out BlockSyntax parentBlock))
             {
                 return;
             }
 
-            var localFunction = (LocalFunctionStatementSyntax)identifier.Parent;
-            if (localFunction.Identifier != identifier)
-            {
-                return;
-            }
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            if (!localFunction.Parent.IsKind(SyntaxKind.Block, out BlockSyntax parentBlock))
+            context.RegisterRefactoring(new MyCodeAction(CSharpFeaturesResources.Convert_to_method,
+                c => UpdateDocumentAsync(root, document, parentBlock, localFunction, c)));
+        }
+
+        private async Task<LocalFunctionStatementSyntax> TryGetSelectedLocalFunctionAsync(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            var localFunction = await CodeRefactoringHelpers.TryGetSelectedNodeAsync<LocalFunctionStatementSyntax>(document, span, cancellationToken).ConfigureAwait(false);
+            if (localFunction == default)
             {
-                return;
+                // Only function's block might be selected -> succeed only when the whole block (all statemets) is selected
```

Comment:
is this a case we actually care about?  i'm struggling to think of someone selecting the body and wanting to use that to mean "convert this local function to a method".  i'm curious why we felt this was necessary?

Reply:
I'd personally expect it to work i.e. it seems to me that from UX perspective there should be no distinction between refactorings offered on method's body and whole method.

Reply:
That seems *really* strange to me.  There's a big difference in my opinion.  If i'm selecting a block, it feels like i'm selecting the *statements* to do something with them.  I'm not selecting the container of the block.  Imagine i have:

```c#
void Foo(int a, int b) [|{
    // stuff
}|]
```

It feels super bizarre to me that the expectation there is that i would be selecting those items just to say that i wanted to move the entire local functino out to a sibling method.

Similarly, if we handle the block case, why do we not handle this case:

```c#
void Foo(int a, int b) [| => Bar(); |]
```

This is also "the method's body".  But i'm really struggling to see why anyone would say "hey i want to convert this whole local function to a method, so i'll select this subportion of it".

I totally understand why they'd select the entire thing.  I totally get why they'd select the name.  I totally get why the care might be anywhere between `void` and even up to the `)` (or at least the `(`). 

But selecting some random portion of the child of it doesn't make sense to me.  No more than selecting this and expecting to get the refactoring:

```c#
void Foo([|int a, int b|]) {
    // stuff
}
```

Reply:
That's a really good point, will update tomorrow accordingly.